audioread/__init__.py

```diff
@@ -70,47 +70,57 @@ def _mad_available():
         return True
 
 
-def audio_open(path):
-    """Open an audio file using a library that is available on this
-    system.
-    """
+def _ffmpeg_available():
+    """Determines whether the ffmpeg program is available."""
+    from . import ffdec
+    return ffdec.available()
+
+
+def available_backends():
+    """Returns a list of backends that are available on this system."""
+
     # Standard-library WAV and AIFF readers.
     from . import rawread
-    try:
-        return rawread.RawAudioFile(path)
-    except DecodeError:
-        pass
+    result = [rawread.RawAudioFile]
 
     # Core Audio.
     if _ca_available():
         from . import macca
-        try:
-            return macca.ExtAudioFile(path)
-        except DecodeError:
-            pass
+        result.append(macca.ExtAudioFile)
 
     # GStreamer.
     if _gst_available():
         from . import gstdec
-        try:
-            return gstdec.GstAudioFile(path)
-        except DecodeError:
-            pass
+        result.append(gstdec.GstAudioFile)
 
     # MAD.
     if _mad_available():
         from . import maddec
+        result.append(maddec.MadAudioFile)
+
+    # FFmpeg.
+    if _ffmpeg_available():
```

Comment:
It's probably fine to just call `ffdec.available()` here instead of making a `_ffmpeg_available` function here. The only reason the `import` is conditional for the MAD backend, for example, is that the import will fail if pymad isn't installed. The ffmpeg backend can be imported anyway (and must be imported to run the check!).

Reply:
It's also worth noting (but not worth worrying about) that this is introducing a small overhead in the one-call case: we now have to invoke ffmpeg twice, once as a backend availability check and once to actually do the decoding. In the old world, we just ran the ffmpeg command once and let it fail if the command isn't available.

Reply:
OK, I'll change it.